### PR TITLE
Ivy integration for ensime search

### PR DESCRIPTION
**Pull request by yyadavalli**
Related to #186 

Basic support for now (goto source only).

Suggestions on more options to add are welcome.

### CODE REVIEW

**Comment by fommil** [comment]
Can we make the ivy dependency optional? Like we do for helm.

File: ensime-search.el [hunk 1]
```diff
@@ -24,6 +24,10 @@
   (require 'ensime-macros))
 
 (require 'dash)
+(require 'ensime-ivy)
```
Comment (fommil) [inline ensime-search.el]: this needs to be opt-in

File: ensime-vars.el [hunk 2]
```diff
@@ -314,9 +314,10 @@ Possible types: `organizeImport', `rename', `extractLocal',
   :type 'boolean
   :group 'ensime-ui)
 
-(defcustom ensime-use-helm nil
-  "Enable helm for some ensime features"
-  :type 'boolean
+(defcustom ensime-search-interface 'classic
```
Comment (fommil) [inline ensime-vars.el]: cool, but will breaks things for some people so we need to tweet / gitter about it.

**Comment by yyadavalli** [comment]
I believe, it's optional. I did not add it to the package file.

**Comment by fommil** [comment]
You can't require it if it's optional... that'll just blow up if the user doesn't have it installed. For helm we require the user to require the ensime-helm package but we could probably be smarter by autoloading it if they've set the appropriate mode.

**Comment by yyadavalli** [comment]
See, if this does the trick. 

This is a breaking change only for `helm` users, an announcement on gitter and a tweet should do the trick.

I can cleanup `ensime-helm.el` and `ensime-search.el`, if you don't mind merging unrelated changes.

**Comment by fommil** [comment]
cool, can you confirm that this works and you don't have helm installed?

**Comment by yyadavalli** [comment]
Yeah, that's how I tested it

**Comment by bomgar** [comment]
The ensime-use-helm variable is still used here https://github.com/ensime/ensime-emacs/blob/master/ensime-editor.el#L729

**Comment by yyadavalli** [comment]
I will check, thanks

**Pull request merged**